Lib/test/test_wsgiref.py

```diff
@@ -525,6 +525,40 @@ def testExtras(self):
             '\r\n'
         )
 
+    validation_cases = (
+        ('Invalid\r\nName', 'ValidValue'),
+        ('Invalid\rName', 'ValidValue'),
+        ('Invalid\nName', 'ValidValue'),
+        ('\r\nInvalidName', 'ValidValue'),
+        ('\rInvalidName', 'ValidValue'),
+        ('\nInvalidName', 'ValidValue'),
+        (' InvalidName', 'ValidValue'),
+        ('\tInvalidName', 'ValidValue'),
+        ('Invalid:Name', 'ValidValue'),
+        (':InvalidName', 'ValidValue'),
+        ('ValidName', 'Invalid\r\nValue'),
+        ('ValidName', 'Invalid\rValue'),
+        ('ValidName', 'Invalid\nValue'),
+        ('ValidName', 'InvalidValue\r\n'),
+        ('ValidName', 'InvalidValue\r'),
+        ('ValidName', 'InvalidValue\n'),
+        (b'InvalidName', 'ValidValue', AssertionError),
+        ('ValidName', b'InvalidValue', AssertionError)
+    )
+
+    def test_add_header_validation(self):
```

Comment:
I suggest to merge the two tests a single test, to factorize code.

I suggest to remove the b'InvalidName' and b'InvalidValue' from validation_cases. Maybe create a second list, maybe put it directly in the test method to avoid `with self.assertRaises(exception[0]) if len(exception) else self.assertRaisesRegex(ValueError, 'Invalid header'): ` complexity.